test/e2e/query_frontend_test.go

```diff
@@ -387,6 +387,27 @@ func TestQueryFrontend(t *testing.T) {
 	})
 }
 
+// https://github.com/thanos-io/thanos/issues/3570
+// "The query splitting test case is flaky because we are using the current timestamp to send queries, which makes the splitting behavior non-deterministic"
+// (Wiard)
+// This should be resolved in a much better way, but for now it removes the flakyness.
+func didItSplitValue(n float64, nightSwitch bool) float64 {
+
+	currentTime := time.Now()
+
+	// Some metrics DO NOT increase if the query range is from 'now -1hour' if it the hour is "0". Then just return the expected value.
+	if nightSwitch && currentTime.Hour() == 0 {
+		return n
+	}
+
+	// For the rest, if it is hour "0" or "1", the query is split, and we should expect double it's values.
+	if currentTime.Hour() == 1 || currentTime.Hour() == 0 {
```

Comment:
Is this dependent on the actual query? :thinking:

Reply:
See my comment above.